### ISSUE TITLE
Debug mode: lock order and position of list of the create unit dialog

### ISSUE THREAD

**Issue by Kwandulin** [wesnoth/wesnoth#2452]
Sometimes it is needed to create several units of the same faction for testing purposes, via debug->create unit. Unfortunately, the position of the list and the recently chosen order (race name, unit name) get set back to default, after quitting the create unit dialog and open it again. One has to scroll through the list again or use the search bar to get back to the units, which is tedious.

I suggest to lock the order and position of the list in the create unit dialog of the debug mode to make life easier

**Comment by Vultraz** [comment]
The last selection is already saved between dialog invocations.

**Comment by Kwandulin** [comment]
Yes, which is a good thing. For testing purposes, it is often helpful to not only get back to the last selected unit, but also to the list entries above and below (e.g. for testing several units of the same faction)

**Comment by Vultraz** [comment]
Alas, that's not so simple.

**Comment by CelticMinstrel** [comment]
What about saving the filter text between invocations? Would that help?

**Comment by Kwandulin** [comment]
Yeah, that would help. It would also help to save the type of ordering (race <-> unit) and whether the list is sorted alphabetically or not. Some further thoughts:
Currently the filter text only applies to the unit type names. Maybe it should apply to either the race or the unit type, depending on which one is sorting the list.
